nemo_aligner/algorithms/dpo.py

```diff
@@ -95,13 +95,16 @@ def __init__(
         self.run_timer = run_timer
 
         self.step = 0
-        self.epoch = 0
         self.consumed_samples = 0
 
         self.ckpt_callback = ckpt_callback
 
-        # used to compute the max step
-        self._train_dataloader_len = len(train_dataloader)
+        # compute `max_steps`
+        sampler = self.train_dataloader.batch_sampler
+        if not sampler.drop_last:
+            raise NotImplementedError("`drop_last=False` is not currently supported")
+        self.num_steps_per_epoch = sampler.total_samples // sampler.global_batch_size
```

Comment:
Could you factorize this into `trainer_utils.py`? So that it becomes something like
`self.num_steps_per_epoch = compute_num_steps_per_epoch(self.train_dataloader.batch_sampler)`

Reply:
Yes, but if you want me to type check the input argument, it would require an annoying, long import from Nemo, and currently `trainer_utils.py` is import lite.

Reply:
This is done, if the imports in `trainer_utils.py` are too much we can revert it

Reply:
Good for me, thanks!